guides/users/custom-authentication.md

```diff
@@ -0,0 +1,172 @@
+# Custom authentication
+
+*You can use the official [`solidus_auth_devise`][solidus-auth-devise] gem
+to provide a `Spree::User` model and basic authentication for Solidus. See its
+documentation for additional setup instructions.*
+
+Stores require a `User` model in order to take advantage of all of Solidus's
+features. This model can have any name, and Solidus can integrate with your
+existing Rails application's existing `User` model.
+
+By default, Solidus provides a [`Spree::LegacyUser` model][legacy-user] that
+offers the bare minimum functionality of a user. The model is intended to be
+modified by extensions.
```

Comment:
This user is only meant as test user and should not be modified by extensions. Also it is not suitable for production environments.

Reply:
Maybe [this](https://github.com/solidusio/solidus/blob/9f5c53aee433fa0e72ae97301a161192e147132e/core/app/models/spree/legacy_user.rb#L4-L5) is just a very old comment then!

```
  # Default implementation of User.
  #
  # @note This class is intended to be modified by extensions (ex.
  #   spree_auth_devise)
```
When I edit this article in my next commits, I'll be sure to update this comment as well.

Reply:
Thanks!